crates/plugin/src/metrics.rs

```diff
@@ -1,8 +1,36 @@
-use std::sync::Arc;
+use std::{sync::Arc, time::Duration};
 
 use log::Level;
 use parking_lot::Mutex;
-use solana_metrics::counter::Counter as CounterInner;
+use solana_metrics::{counter::Counter as CounterInner, datapoint_info};
+
+use crate::message_processor::QUEUE_DEPTH_REPORT_INTERVAL;
+
+/// A very simple metric that reports a single value
+#[derive(Debug)]
+pub struct GaugeMetric {
+    pub name: &'static str,
+    last_reported_instant: Mutex<std::time::Instant>,
+    report_interval: Duration,
+}
+
+impl GaugeMetric {
+    pub fn new(name: &'static str, report_interval: Duration) -> Self {
+        Self {
```

Comment:
could actually encapsulate a thread that logs this out every duration here.  mentioned above, kinda same applies.  point is to clear the hot path